libusb/core.c

```diff
@@ -2452,14 +2468,22 @@ int API_EXPORTED libusb_init_context(libusb_context **ctx, const struct libusb_i
 		if (LIBUSB_OPTION_LOG_LEVEL == option || !default_context_options[option].is_set) {
 			continue;
 		}
-		r = libusb_set_option(_ctx, option);
+		if (LIBUSB_OPTION_LOG_CB != option) {
+			r = libusb_set_option(_ctx, option);
+		} else {
+			r = libusb_set_option(_ctx, option, default_context_options[option].arg.log_cbval);
+		}
 		if (LIBUSB_SUCCESS != r)
 			goto err_free_ctx;
 	}
 
 	/* apply any options provided by the user */
 	for (int i = 0 ; i < num_options ; ++i) {
-		r = libusb_set_option(_ctx, options[i].option, options[i].value.ival);
+		if (LIBUSB_OPTION_LOG_CB != options[i].option) {
+			r = libusb_set_option(_ctx, options[i].option, options[i].value.ival);
+		} else {
+			r = libusb_set_option(_ctx, options[i].option, options[i].value.log_cbval);
+		}
```

Comment:
assuming more options might appear in the future, I'd suggest make this a `switch`, e.g.:
```suggestion
		switch (options[i].option)
		case LIBUSB_OPTION_LOG_CB:
			r = libusb_set_option(_ctx, options[i].option, options[i].value.log_cbval);
			break;
		default:
			r = libusb_set_option(_ctx, options[i].option, options[i].value.ival);
			break;
		}
```
I find this way a bit cleaner for understanding/reading

Reply:
Commit suggestion is not working for some reason so did this by hand.